src/basic/user-util.c

```diff
@@ -720,10 +722,18 @@ bool valid_user_group_name_full(const char *u, bool strict) {
 
         if (!(u[0] >= 'a' && u[0] <= 'z') &&
             !(u[0] >= 'A' && u[0] <= 'Z') &&
+            !(u[0] >= '0' && u[0] <= '9') &&
             u[0] != '_')
                 return false;
 
-        bool warned = false;
+        if (u[0] >= '0' && u[0] <= '9') {
+                if (strict)
+                        return false;
+                else {
+                        log_warning("Bad user or group name \"%s\", accepting for compatibility.", u);
```

Comment:
I'd prefer a more precise error message: "User or group name starts with a digit, accepting for compatibility: %s" or something like that.

Reply:
Fixed.